lib/jsonapi/query_parser.ex

```diff
@@ -194,4 +198,16 @@ defmodule JSONAPI.QueryParser do
     view = Keyword.fetch!(opts, :view)
     struct(JSONAPI.Config, opts: opts, view: view)
   end
+
+  defp struct_from_map(params, struct) do
+    struct_keys = Map.keys(struct) |> Enum.filter(fn x -> x != :__struct__ end)
```

Comment:
https://hexdocs.pm/elixir/Map.html#from_struct/1 Map.from_struct will accomplish this.